src/hydrovu/client.rs

```diff
@@ -0,0 +1,111 @@
+use std::marker::PhantomData;
+use std::rc::Rc;
+use super::error::Error;
+use super::constant;
+use anyhow::Context;
+
+use oauth2::{
+    basic::BasicClient, reqwest::http_client, AuthUrl, ClientId, ClientSecret, Scope,
+    TokenResponse, TokenUrl,
+};
+
+pub struct Client {
+    client: reqwest::blocking::Client,
+    token: String,
+}
+
+pub struct ClientCall<T: for<'de> serde::Deserialize<'de>> {
+    client: Rc<Client>,
+    url: String,
+    next: Option<String>,
+    phan: PhantomData<T>,
+}
+
+impl Client {
+    pub fn fetch_json<T: for<'de> serde::Deserialize<'de>>(client: Rc<Client>, url: String) -> ClientCall<T> {
+	ClientCall::<T> {
+            client: client,
+            url: url,
+            next: Some("".to_string()),
+            phan: PhantomData,
+	}
+    }
```

Comment:
@e-dard after refactoring and placing things in separate modules, it ends up here.